api/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -1394,6 +1408,19 @@ type GlobalConfig struct {
 	// Default: 'true'
 	// +optional
 	RunProcessChecksInCoreAgent *bool `json:"runProcessChecksInCoreAgent,omitempty"`
+
+	// CoreAgent contains the settings for the core agent.
+	// +optional
+	CoreAgent *CoreAgent `json:"coreAgent,omitempty"`
```

Comment:
This config doesn't really enabled/disable core agent. The value is just added as an env var to bunch of components. 
What is this configuration for? How would customer determine whether to set it to true or false?